### PR TITLE
xds: allow grpclb balancer addresses for backward compatibility

### PR DESCRIPTION
**Pull request by dapengzhang0**
During migration, the name resolver may not know when the client has been upgraded to xds, so it may still send grpclb v1 addresses with a list of policies including both grpclb v1 and xds.

### CODE REVIEW

**Comment by zhangkun83** [comment]
This work is done in the XdsLoadBalancer -- doesn't it require client to be upgraded to xDS in the first place?

File: xds/src/main/java/io/grpc/xds/XdsLoadBalancer.java [hunk 1]
```diff
@@ -326,14 +328,24 @@ void updateFallbackServers(
         List<EquivalentAddressGroup> servers, Attributes attributes,
         LbConfig fallbackPolicy) {
       this.fallbackServers = servers;
+      String fallbackPolicyName = fallbackPolicy.getPolicyName();
+      if (!fallbackPolicyName.equals("grpclb") && !fallbackPolicyName.equals(XDS_POLICY_NAME)) {
```
Comment (zhangkun83) [inline xds/src/main/java/io/grpc/xds/XdsLoadBalancer.java]: Probably worth a comment here.

Comment (dapengzhang0) [reply xds/src/main/java/io/grpc/xds/XdsLoadBalancer.java]: Done.

File: xds/src/main/java/io/grpc/xds/XdsLoadBalancer.java [hunk 2]
```diff
@@ -360,6 +340,38 @@ void updateFallbackServers(
...
+        ImmutableList.Builder<EquivalentAddressGroup> backends = ImmutableList.builder();
+        for (EquivalentAddressGroup eag : fallbackServers) {
+          if (eag.getAttributes().get(GrpcAttributes.ATTR_LB_ADDR_AUTHORITY) == null) {
+            backends.add(eag);
+          }
+        }
+        servers = backends.build();
+      }
+
+      // FIXME: this is a temporary hack.
```
Comment (zhangkun83) [inline xds/src/main/java/io/grpc/xds/XdsLoadBalancer.java]: Please add TODO(zhangkun83) and refer to https://github.com/grpc/grpc-java/issues/5496

File: xds/src/main/java/io/grpc/xds/XdsLoadBalancer.java [hunk 3]
```diff
@@ -360,6 +340,38 @@ void updateFallbackServers(
       }
     }
 
+    private void handleFallbackAddressesOrNameResolutionError() {
```
Comment (zhangkun83) [inline xds/src/main/java/io/grpc/xds/XdsLoadBalancer.java]: Why does the method name contain "NameResolutionError"? It's only called from fallback-related methods. What this method does is passing the latest fallback addresses to the fallback balancer. The handling of empty addresses is just part of it. Maybe "propagateFallbackAddresses()".

File: xds/src/main/java/io/grpc/xds/XdsLoadBalancer.java (hunk 2 above)
Comment (dapengzhang0) [reply xds/src/main/java/io/grpc/xds/XdsLoadBalancer.java]: Done.

File: xds/src/main/java/io/grpc/xds/XdsLoadBalancer.java (hunk 3 above)
Comment (dapengzhang0) [reply xds/src/main/java/io/grpc/xds/XdsLoadBalancer.java]: Renamed

**Pull request merged**